### PR TITLE
Concept renaming

### PR DESCRIPTION
**Pull request by MitraDarja**
alphabet_tuple_base_concept -> AlphabetTupleBase
validator_concept -> Validator
char_concept -> Char

### CODE REVIEW

File: include/seqan3/alphabet/adaptation/concept.hpp [hunk 1]
```diff
@@ -44,11 +44,11 @@ namespace seqan3
  */
 //!\cond
 template <typename type>
-SEQAN3_CONCEPT char_adaptation_concept = Alphabet<type> &&
+SEQAN3_CONCEPT CharAdaptationAlphabet = Alphabet<type> &&
```
Comment (eseiler) [inline include/seqan3/alphabet/adaptation/concept.hpp]: Can you also fix the indent here? (also for the other concepts in this file)

**Comment by h-2** [comment]
I am currently doing major refactoring in the alphabet module and will likely fix things on the way. I think some fo the concepts can be removed anyway.

File: include/seqan3/alphabet/adaptation/concept.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * The requirements for this concept are given as related functions and metafunctions.
+ * Types that satisfy this concept are shown as "implementing this interface".
+ */
+//!\cond
+template <typename type>
+SEQAN3_CONCEPT UintAdaptationAlphabet = Alphabet<type> &&
+                                        detail::is_uint_adaptation_v<type>;
+//!\endcond
+
+} // namespace seqan3
```
Comment (h-2) [inline include/seqan3/alphabet/adaptation/concept.hpp]: Why did you re-add this file after it was removed?

File: include/seqan3/argument_parser/all.hpp [hunk 3]
```diff
@@ -32,7 +32,11 @@
  * # Argument Validation
  *
  * The SeqAn 3 Argument Parser offers a validation mechanism for (positional_)options
+<<<<<<< HEAD
  * via callables. You can pass any functor that fulfils the seqan3::validator_concept
+=======
+ * via callables. You can pass any functor that fulfills the seqan3::Validator
+>>>>>>> eafa0199... [MISC] validator_concept to Validator
```
Comment (h-2) [inline include/seqan3/argument_parser/all.hpp]: There are unresolved merge conflicts here!

File: include/seqan3/core/concept/core_language.hpp [hunk 4]
```diff
@@ -124,16 +124,16 @@ template <typename t>
...
  * \brief       This concept encompasses exactly the types `char`, `signed char`, `unsigned char`, `wchar_t`,
  *              `char16_t` and `char32_t`.
  */
 //!\cond
 
 template <typename t>
-SEQAN3_CONCEPT char_concept = std::Integral<t> &&
-                       (std::Same<t, char> || std::Same<t, unsigned char> || std::Same<t, signed char> ||
+SEQAN3_CONCEPT Char = std::Integral<t> &&
+                      (std::Same<t, char> || std::Same<t, unsigned char> || std::Same<t, signed char> ||
```
Comment (h-2) [inline include/seqan3/core/concept/core_language.hpp]: also correct indention below

File: include/seqan3/range/view/to_lower.hpp [hunk 5]
```diff
@@ -66,8 +66,8 @@ namespace seqan3::view
  */
 inline auto const to_lower = deep{std::view::transform([] (auto const in) noexcept
 {
-    static_assert(char_concept<remove_cvref_t<decltype(in)>>,
-                  "The value type of seqan3::view::to_lower must model the seqan3::char_concept.");
+    static_assert(Char<remove_cvref_t<decltype(in)>>,
+                  "The value type of seqan3::view::to_lower must model the seqan3::Char.");
```
Comment (h-2) [inline include/seqan3/range/view/to_lower.hpp]:
```suggestion
                  "The value type of seqan3::view::to_lower must model seqan3::Char.");
```

**Comment by h-2** [comment]
Approved, but has conflicts.

**Comment by MitraDarja** [comment]
@h-2 Conflicts should be gone now. :)

**Comment by h-2** [comment]
Hm, either they werent fixed, or they got broken again. Can you update once more? Thank you!

**Comment by MitraDarja** [comment]
@h-2 Another try, it should be fixed.

**Comment by h-2** [comment]
Oh no. A merge conflict again. Also I am not sure why you need the commit removing lemon. It's already gone. Can you do another rebase and also try without the lemon commit? @marehr has promised to help debug your git if it doesn't work.

**Pull request merged**